### PR TITLE
Expanding preconditions checks description to help 3p developers

### PR DESCRIPTION
**Pull request by ankitaj224**

### CODE REVIEW

File: firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java [hunk 1]
```diff
@@ -143,9 +143,12 @@ public Thread newThread(Runnable r) {
...
   private void preConditionChecks() {
-    Preconditions.checkNotEmpty(getApplicationId());
-    Preconditions.checkNotEmpty(getProjectIdentifier());
-    Preconditions.checkNotEmpty(getApiKey());
+    Preconditions.checkNotEmpty(getApplicationId(), "Firebase App ID should not be empty or null.");
+    Preconditions.checkNotEmpty(
+        getProjectIdentifier(), "Firebase project ID should not be empty or null.");
+    Preconditions.checkNotEmpty(
+        getApiKey(),
+        "Firebase API key used for authenticating requests from your app should not be empty or null.");
```
Comment (andirayo) [inline firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java:151]: Please set a valid API key. A Firebase API key is required to communicate with Firebase server APIs: It authenticates your project with Google.

File: firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java [hunk 2]
```diff
@@ -143,9 +143,12 @@ public Thread newThread(Runnable r) {
    * or empty.
    */
   private void preConditionChecks() {
-    Preconditions.checkNotEmpty(getApplicationId());
-    Preconditions.checkNotEmpty(getProjectIdentifier());
-    Preconditions.checkNotEmpty(getApiKey());
+    Preconditions.checkNotEmpty(getApplicationId(), "Firebase App ID should not be empty or null.");
+    Preconditions.checkNotEmpty(
+        getProjectIdentifier(), "Firebase project ID should not be empty or null.");
```
Comment (andirayo) [inline firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java:148]: Please set your project ID. A valid Firebase project ID is required to communicate with Firebase server APIs: It identifies your project with Google.

File: firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java [hunk 3]
```diff
@@ -143,9 +143,12 @@ public Thread newThread(Runnable r) {
    * or empty.
    */
   private void preConditionChecks() {
-    Preconditions.checkNotEmpty(getApplicationId());
-    Preconditions.checkNotEmpty(getProjectIdentifier());
-    Preconditions.checkNotEmpty(getApiKey());
+    Preconditions.checkNotEmpty(getApplicationId(), "Firebase App ID should not be empty or null.");
```
Comment (andirayo) [inline firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java:146]: Please set your Application ID. A valid Firebase App ID is required to communicate with Firebase server APIs: It identifies your application with Firebase.

**Comment by ankitaj224** [comment]
> Thank you! Looks good.
> I made a couple of comments.
> 
> One thing I am generally worried about is this:
> If the exception thrown by `Preconditions.checkNotEmpty` gets swallowed by a try-catch block, the developer will never see this exception message.
> Is it possible to also print to the logs?

I tested it with FIS test app and it clearly logged the reason. The preconditions check results in task failure and that gets printed in logs. 

Adding additional logs will add noise to the code and mostly will be rework of the preconditions check.

**Pull request merged**